<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LineWindowClass</class>
 <widget class="QMainWindow" name="LineWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1242</width>
    <height>890</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LineWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>609</width>
      <height>815</height>
     </rect>
    </property>
    <property name="title">
     <string>显示窗口</string>
    </property>
    <widget class="CMyLabel" name="lbl_Window">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>571</width>
       <height>501</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtn_ReadImage">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>打开图片</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtn_FitImage">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>自适应</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_Status">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>550</y>
       <width>561</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>坐标灰度信息</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ClearWindow">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>清空窗口</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="ckb_ShowObj">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>600</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>缩放时显示图形</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_Hide">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>关闭</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>624</x>
      <y>9</y>
      <width>609</width>
      <height>815</height>
     </rect>
    </property>
    <property name="title">
     <string>操作窗口</string>
    </property>
    <widget class="QTabWidget" name="tabWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>571</width>
       <height>771</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_11">
      <attribute name="title">
       <string>卡尺设置</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label_38">
             <property name="text">
              <string>起点Row</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_RowBegin_Line">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.000100000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_39">
             <property name="text">
              <string>起点Col</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_ColBegin_Line">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.000100000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_40">
             <property name="text">
              <string>终点Row</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_RowEnd_Line">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.000100000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_41">
             <property name="text">
              <string>终点Col</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_ColEnd_Line">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.000100000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_31">
             <property name="text">
              <string>卡尺数量</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spb_MeasureNum_Line">
             <property name="maximum">
              <number>999999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="label_29">
             <property name="text">
              <string>搜索长度</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spb_MeasureLength1_Line">
             <property name="maximum">
              <number>999999</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="label_30">
             <property name="text">
              <string>投影长度</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spb_MeasureLength2_Line">
             <property name="maximum">
              <number>999999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QLabel" name="label_32">
             <property name="text">
              <string>平滑(Sigma)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_MeasureSigma_Line">
             <property name="decimals">
              <number>1</number>
             </property>
             <property name="minimum">
              <double>0.400000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QLabel" name="label_33">
             <property name="text">
              <string>阈值</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spb_MeasureThreshold_Line">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>20</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <widget class="QLabel" name="label_34">
             <property name="text">
              <string>极性</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_MeasureTransition_Line">
             <item>
              <property name="text">
               <string>positive</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>negative</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>all</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>uniform</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <widget class="QLabel" name="label_35">
             <property name="text">
              <string>边缘选取</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_MeasureSelect_Line">
             <item>
              <property name="text">
               <string>first</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>last</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>all</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QLabel" name="label_36">
             <property name="text">
              <string>最小得分</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_MinScore_Line">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>0.550000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_17">
           <item>
            <widget class="QLabel" name="label_47">
             <property name="text">
              <string>距离剔除</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dspb_DistanceThreshold_Line">
             <property name="decimals">
              <number>1</number>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>3.500000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_18">
           <item>
            <widget class="QLabel" name="label_37">
             <property name="text">
              <string>插值算法</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_MeasureInterpolation_Line">
             <item>
              <property name="text">
               <string>nearest_neighbor</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>bilinear</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>bicubic</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="QLabel" name="label_49">
             <property name="text">
              <string>是否跟随</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_IsFollow_Line">
             <item>
              <property name="text">
               <string>No</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Yes</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>272</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QPushButton" name="btn_DrawLine_Line">
           <property name="text">
            <string>绘制直线</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_DetectLine_Line">
           <property name="text">
            <string>检测直线</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_ClearRoi_Line">
           <property name="text">
            <string>清除ROI</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_12">
      <attribute name="title">
       <string>结果</string>
      </attribute>
      <widget class="QWidget" name="">
       <property name="geometry">
        <rect>
         <x>14</x>
         <y>27</y>
         <width>531</width>
         <height>641</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QPushButton" name="btn_SaveLineData">
              <property name="text">
               <string>参数保存</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_LoadLineData">
              <property name="text">
               <string>参数加载</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="lbl_RunLineCT">
            <property name="text">
             <string>运行时间</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_6">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>288</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_11">
            <item>
             <widget class="QLabel" name="label_50">
              <property name="text">
               <string>起点X:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_51">
              <property name="text">
               <string>起点Y:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_52">
              <property name="text">
               <string>终点X:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_53">
              <property name="text">
               <string>终点Y:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12">
            <item>
             <widget class="QLineEdit" name="txt_Col1_Line">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="txt_Row1_Line">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="txt_Col2_Line">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="txt_Row2_Line">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_13">
      <attribute name="title">
       <string>点结果</string>
      </attribute>
      <widget class="QTableWidget" name="tablewidget_PointsResult_Line">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>561</width>
         <height>671</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1242</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CMyLabel</class>
   <extends>QLabel</extends>
   <header>cmylabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
